course/test/src/Environment_TEST.cc

```diff
@@ -0,0 +1,86 @@
+#include <cmath>
+#include <sstream>
+#include <string>
+#include <memory>
+
+#include "gtest/gtest.h"
+
+#include "environment.h"
+#include "ant.h"
+#include "cell_mock.h"
+
+namespace ekumen {
+namespace simulation {
+namespace test {
+namespace {
+
+using cell_ptr = std::shared_ptr<Cell>;
+
+GTEST_TEST(EnvironmentTest, ThrowsExceptionOnConstructorWithNonPositiveRows) {
+    ASSERT_THROW(Environment env(0, 1, 0, 0), std::invalid_argument);
+    ASSERT_THROW(Environment env(-1, 1, 0, 0), std::invalid_argument);
+}
+
+GTEST_TEST(EnvironmentTest, ThrowsExceptionOnConstructorWithNonPositiveCols) {
+    ASSERT_THROW(Environment env(1, 0, 0, 0), std::invalid_argument);
+    ASSERT_THROW(Environment env(1, -1, 0, 0), std::invalid_argument);
+}
+
+GTEST_TEST(EnvironmentTest, ConstructorWithPositiveDimensionsDontThrowException) {
+    ASSERT_NO_THROW(Environment env(1, 1, 0, 0));
+}
+
+GTEST_TEST(EnvironmentTest, ConstructorWithMoreInsectsThanCellsThrowsException) {
+    ASSERT_THROW(Environment env(1, 1, 2, 2), std::invalid_argument);
+}
+
+GTEST_TEST(EnvironmentTest, SimulationFinishedWhenCreatedWithNoInsect) {
+    Environment env(1, 1, 0, 0);
+    EXPECT_TRUE(env.SimulationFinished());
+}
+
+GTEST_TEST(EnvironmentTest, SimulationNotFinishedWhenCreatedInsects) {
+    Environment env(2, 2, 1, 1);
+    EXPECT_FALSE(env.SimulationFinished());
+}
+
+GTEST_TEST(EnvironmentTest, SimulationFinishedWhenThereIsASingleCellAndASingleInsect) {
+    Environment env1(1, 1, 0, 1);
+    EXPECT_TRUE(env1.SimulationFinished());
+    Environment env2(1, 1, 0, 1);
```

Comment:
This is the same case as above

Reply:
You're right. I just wanted to make sure that the condition applies both when you start with only an Ant and only a Doodlebug.